docs/source/configuration/validation.md

```diff
@@ -0,0 +1,197 @@
+You should specify the `type` in the JSON schema of the block (in a content type, this is included in the default serialization of the field).
+The next example is for the use case of a block JSON schema:
+
+```ts
+let blockSchema = {
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom field',
+      description: '',
+      type: 'integer'
+    },
+  },
+  required: [],
+};
+```
+
+### Per field `widget` validators
+
+These validators are applied depending on the specified `widget` of the field.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['myCustomURLWidget', 'maximum'],
+  component: maximumValidator,
+});
+```
+
+It takes two `dependencies` since we can potentially have several validators for the same `widget`.
+The first element should be the `widget`, and the second you can set it up to identify the validator.
+You should specify the `widget` in the JSON schema of the block (or as additional data in the content type definition).
+The next example is for the use case of a block JSON schema:
+
+```ts
+let blockSchema = {
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom field',
+      description: '',
+      widget: 'myCustomURLWidget',
+    },
+  },
+  required: [],
+};
+```
+
+### Per behavior and field name validator
+
+These validators are applied depending on the behavior (usually coming from a content type definition) in combination with the name of the field.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['plone.eventbasic', 'start'],
+  component: urlValidator,
+});
+```
+The first dependency should be the name of the behavior, and the second the name (`id`) of the field.
+This type of validator only applies to content-type validators.
+
+### Per block type and field name validator
+
+These validators are applied depending on the block type in combination with the name of the field in the block settings JSON schema.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['slider', 'url'],
+  component: urlValidator,
+});
+```
+The first dependency should be the `id` of the block, and the second the `id` of the field.
+This type of validator only applies to blocks.
+
+### Specific validator using the `validator` key in the field
+
+A final type of validator is applied to the field if the `validator` key is present in the JSON schema definition of the form field.
+
+```ts
+config.registerComponent({
+  name: 'fieldValidator',
+  dependencies: ['isURL'],
+  component: urlValidator,
+});
+```
+
+The dependencies take one single name, in this case, the name of the validator.
+You should specify the validator in the JSON schema of the block (or as additional data in the content type definition).
+
+```ts
+let blockSchema = {
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'Default field',
+      description: '',
+      validator: 'isURL',
```

Comment:
Maybe allow passing directly a validator function as well? We're way past the point "schemas should be serializable" and for one-offs it's easier to declare them inline.

Reply:
If the pattern we are teaching is go for "named" components and call them by name (partially because of the duality forms from the backend/content types and the client defined forms in blocks), I'd refrain to do so, because then we will have two use cases, one where you can do it, and another use case in which you can't. 

Also, in terms of explaining this in docs will became tricky and separate this is already confusing (I've been there when reviewing this with @ericof). Some other considerations is testability, because in that regard, modularization >> inlining things.

So, in principle, I would be -1 for this.